app/src/main/java/com/hover/stax/SplashScreenActivity.java

```diff
@@ -30,29 +40,74 @@
 	@Override
 	protected void onCreate(@Nullable Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
+		startSplashForegroundSequence();
+		startBackgroundProcesses();
+	}
 
-//		if (SelfDestructActivity.isExpired(this)) {
-//			startActivity(new Intent(this, SelfDestructActivity.class));
-//			finish();
-//			return;
-//		}
+	private void startSplashForegroundSequence() {
+		initSplashAnimation();
+		authenticateBiometricOrNavigateScreenAfter110sec();
+	}
 
-		Amplitude.getInstance().initialize(this, getString(R.string.amp)).enableForegroundTracking(getApplication());
+	private void startBackgroundProcesses() {
+		initAmplitude();
 		initHover();
 		createNotificationChannel();
 		startWorkers();
+	}
 
-		if (Utils.getSharedPrefs(this).getInt(AUTH_CHECK, 0) == 1)
-			new BiometricChecker(this, this).startAuthentication(null);
-		else
-			chooseNav();
+	private void initSplashAnimation() {
+		setContentView(R.layout.splash_screen_layout);
+		blurBackgroundAfter60sec();
+		fadeInSplashContentAfter90sec();
+	}
+
+	private void blurBackgroundAfter60sec() {
+		new Handler(Looper.getMainLooper()).postDelayed(() -> {
+			Bitmap bg = BitmapFactory.decodeResource(getResources(), R.drawable.stax_splash);
+			Bitmap bitmap = new StaxBlur(this,16, 1).transform(bg);
+			ImageView bgView = findViewById(R.id.splash_image_blur);
+			if (bgView != null) {
+				bgView.setImageBitmap(bitmap);
+				bgView.setVisibility(View.VISIBLE);
+				bgView.setAnimation(UIHelper.loadFadeIn(this));
+			}
+		}, 1000);
+	}
+
+	private void fadeInSplashContentAfter90sec() {
+		TextView tv = findViewById(R.id.splash_content);
+		new Handler(Looper.getMainLooper()).postDelayed(() -> {
+			tv.setVisibility(View.VISIBLE);
+			tv.setAnimation(UIHelper.loadFadeIn(this));
+		}, 1200);
+	}
+
+	private void authenticateBiometricOrNavigateScreenAfter110sec() {
+		new Handler().postDelayed(() -> {
+			if (Utils.getSharedPrefs(this).getInt(AUTH_CHECK, 0) == 1) new BiometricChecker(this, this).startAuthentication(null);
+			else chooseNav();
+		}, 1800);
+
+	}
+
+	private void initAmplitude() {
+		Amplitude.getInstance().initialize(this, getString(R.string.amp)).enableForegroundTracking(getApplication());
 	}
 
 	private void initHover() {
 		Hover.initialize(this);
 		Hover.setBranding(getString(R.string.app_name), R.mipmap.stax, this);
 	}
 
+	private Boolean shouldSelfDestructWhenAppVersionExpires(Boolean value) {
```

Comment:
Not sure how this works, but you can look at Firebase Remote Configs for this - would be much simpler